src/d/d_wood.cpp

```diff
@@ -4,184 +4,899 @@
+  } else {
+    return false;
+  }
 }
 
 /* 800BE93C-800BEA28       .text set_mtx__Q25dWood6Unit_cFPQ25dWood5Anm_c */
-void dWood::Unit_c::set_mtx(dWood::Anm_c*) {
-    /* Nonmatching */
-}
+void dWood::Unit_c::set_mtx(dWood::Anm_c *anim) {
+  int iVar1;
 
-/* 800BEA28-800BEA50       .text clear__Q25dWood6Unit_cFv */
-void dWood::Unit_c::clear() {
-    /* Nonmatching */
-}
+  iVar1 = mAnmIdx;
 
-/* 800BEA50-800BEE9C       .text cc_hit_before_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
-void dWood::Unit_c::cc_hit_before_cut(dWood::Packet_c*) {
-    /* Nonmatching */
-}
+  mDoMtx_copy(anim[iVar1].mModelMtx, mDoMtx_stack_c::get());
+  mDoMtx_stack_c::now[0][3] = mDoMtx_stack_c::now[0][3] + (mPos).x;
+  mDoMtx_stack_c::now[1][3] = mDoMtx_stack_c::now[1][3] + (mPos).y;
+  mDoMtx_stack_c::now[2][3] = mDoMtx_stack_c::now[2][3] + (mPos).z;
+  mDoMtx_concat(j3dSys.getViewMtx(), mDoMtx_stack_c::now, mModelViewMtx);
+
+  mDoMtx_copy(anim[iVar1].mTrunkModelMtx, mDoMtx_stack_c::get());
+  mDoMtx_stack_c::now[0][3] = (mPos).x;
+  mDoMtx_stack_c::now[1][3] = (mPos).y;
+  mDoMtx_stack_c::now[2][3] = (mPos).z;
+  mDoMtx_concat(j3dSys.getViewMtx(), mDoMtx_stack_c::get(),
+                mTrunkModelViewMtx);
 
-/* 800BEE9C-800BEEA0       .text cc_hit_after_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
-void dWood::Unit_c::cc_hit_after_cut(dWood::Packet_c*) {
-    /* Nonmatching */
+  mDoMtx_concat(j3dSys.getViewMtx(), mShadowModelMtx,
+                mShadowModelViewMtx);
 }
 
+/* 800BEA28-800BEA50       .text clear__Q25dWood6Unit_cFv */
+void dWood::Unit_c::clear() { cLib_memSet(this, 0, 0x18c); }
+
+/* 800BEA50-800BEE9C       .text
+ * cc_hit_before_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
+void dWood::Unit_c::cc_hit_before_cut(dWood::Packet_c *packet) {
+  int animIdx;
+  int oldAnimIdx;
+
+  dCcMassS_HitInf inf;
+  fopAc_ac_c *actor;
+  u32 ret = dComIfG_Ccsp()->ChkMass(&mPos, &actor, &inf);
+
+  if (mAnimCooldown > 0) {
+    mAnimCooldown -= 1;
+  }
+
+  // Evaluate for attacks that will not cut us down
+  if ((ret & 1)) {
+    cCcD_Obj *atHitObj = inf.GetAtHitObj();
+    if (atHitObj != NULL &&
+        // @TODO: I think the cCcD_ObjAtType enum is reversed. E.g. the first
+        // one should be Machete (bit 21) not Wind
+        (atHitObj->ChkAtType(AT_TYPE_WIND) ||
+         atHitObj->ChkAtType(AT_TYPE_BOMB) ||
+         atHitObj->ChkAtType(AT_TYPE_FIRE) ||
+         atHitObj->ChkAtType(AT_TYPE_NORMAL_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_FIRE_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_ICE_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_LIGHT_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_HOOKSHOT))) {
+
+      // Clear the hit bit so that we don't get cut down
+      ret &= ~0x01;
+
+      if (actor != NULL && mAnimCooldown == 0) {
+        animIdx = packet->search_anm(Anm_c::Mode_PushInto);
+        mAnimCooldown = 20;
+
+        // Play the cut sound
+        JAIZelBasic::getInterface()->seStart(JA_SE_OBJ_TREE_SWING, &mPos,
```

Comment:
this is the mDoAud_seStart inline, same for the other seStart calls here

Reply:
Oo gotcha. Where do the mDo* inline functions come from? Are they just patterns that you see in multiple files and decide to make a mDo* function for? 

Is there any easy way to spot them in Ghidra?

Reply:
inline functions are official from the original devs, we don't add any they didn't make (except in *very* rare cases where it's required to match a function).

there's no way for ghidra to notice them because we only have a release build of the game, which has them all inlined away. but you can check the kiosk demo debug maps (pinned in discord) and they list all of the inlines in the game (except ones that don't exist in the demo but do in the final game, but those are also rare).

Reply:
Ohhhhh that's so good to know